### ISSUE TITLE
Stricter checking/validation of statefulset pvcs

### ISSUE THREAD

**Issue by bprashanth** [kubernetes/kubernetes#41153]
Currently theren's no enforcement around statefulset volume templates, which means the burden is on the user to make sure there's no collision. A user can create:

Petset 1: petset  Claim: alpha-beta
Petset 2: beta-petset Claim: alpha

and end up with a collision. The controller should be smarter, maybe we need something like parent-ref? 
@kubernetes/sig-apps-bugs @jingxu97

**Comment by smarterclayton** [comment]
I'm actually wondering whether this might be valuable in some circumstances.

**Comment by bprashanth** [comment]
Yeah it does allow sharing, and we can assume the user has set the right access modes, but I was thinking the risk of wrong access mode x unintentional sharing is higher than other places where we tolerate such issues (eg: overlapping selectors)

**Comment by smarterclayton** [comment]
Any reports of this happening yet?  I could see setting parent ref, but we still aren't clear on whether deleting the SS should orphan the pvc the same way.  I want to orphan pvc differently than pods in many cases

**Comment by kow3ns** [comment]
No reports of collisions afaik, but the naming scheme causes issues with the internal hashing mechanics of dynamic provisioner when it is trying to ensure that all claims for a Pod in an SS are met by volumes allocated in the same zone/region #40910.

**Comment by justinsb** [comment]
@kow3ns yes - that's how we/I found the issue!  #40910 will be fixed real-soon-now though - that's just a silly bug.
